tests/test_copy_novaseqx_flow_cell.py

```diff
@@ -0,0 +1,180 @@
+) -> Path:
+    Path(flow_cell_directory, DemultiplexingDirsAndFiles.ANALYSIS, latest_analysis_version).mkdir(
+        parents=True
+    )
+    Path(
+        flow_cell_directory,
+        DemultiplexingDirsAndFiles.ANALYSIS,
+        latest_analysis_version,
+        DemultiplexingDirsAndFiles.COPY_COMPLETE,
+    ).touch()
+    return flow_cell_directory
+
+
+@pytest.fixture
+def demultiplex_not_complete_novaseqx_flow_cell(tmp_file: Path) -> Path:
+    return tmp_file
+
+
+def test_flow_cell_is_ready_for_post_processing(novaseqx_flow_cell: Path, demultiplexed_runs: Path):
+    # GIVEN a flow cell which is ready for post processing
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(novaseqx_flow_cell, demultiplexed_runs)
+
+    # THEN the flow cell is ready
+    assert ready
+
+
+def test_is_not_ready_without_analysis(
+    novaseqx_flow_cell_analysis_incomplete: Path, demultiplexed_runs: Path
+):
+    # GIVEN a flow cell for which analysis is not completed
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(novaseqx_flow_cell_analysis_incomplete, demultiplexed_runs)
+
+    # THEN it is not ready
+    assert not ready
+
+
+def test_flow_cell_is_not_demultiplexed(
+    demultiplex_not_complete_novaseqx_flow_cell: Path, demultiplexed_runs: Path
+):
+    # GIVEN a flow cell for which demultiplexing is not completed
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        demultiplex_not_complete_novaseqx_flow_cell, demultiplexed_runs
+    )
+
+    # THEN it is not ready
+    assert not ready
+
+
+def test_previously_post_processed_flow_cell_is_not_ready(
+    post_processed_novaseqx_flow_cell: Path, demultiplexed_runs: Path
+):
+    # GIVEN a flow cell for which post processing is done
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(post_processed_novaseqx_flow_cell, demultiplexed_runs)
+
+    # THEN the flow cell is not ready
+    assert not ready
+
+
+def test_previously_copied_flow_cell_is_not_ready(
+    novaseqx_flow_cell: Path, demultiplexed_runs: Path
+):
+    # GIVEN a flow cell which already exists in demultiplexed runs
+    Path(demultiplexed_runs, novaseqx_flow_cell.name).mkdir()
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(novaseqx_flow_cell, demultiplexed_runs)
+
+    # THEN the flow cell is not ready
+    assert not ready
+
+
+def test_get_latest_analysis_version_path(
+    novaseqx_flow_cell: Path,
+    latest_analysis_version: str,
+):
+    # GIVEN a flow cell which is ready to be post processed
+
+    # WHEN extracting the latest analysis version path
+    analysis = get_latest_analysis_directory(novaseqx_flow_cell)
+
+    # THEN the latest analysis version path is returned
+    latest_analysis = Path(
```

Comment:
```suggestion
    latest_analysis: Path = Path(
```

Reply:
I don't see the value of adding type hints for variables where the object is directly instantiated